pkg/kind/kong_stream_listen.go

```diff
@@ -0,0 +1,232 @@
+		if err != nil {
+			return
+		}
+		// the deployment itself needs to be ready
+		if d.Status.ReadyReplicas == d.Status.Replicas && d.Status.AvailableReplicas == d.Status.Replicas && d.Status.UnavailableReplicas < 1 {
+			var proxySVC *corev1.Service
+			proxySVC, err = cluster.Client().CoreV1().Services(proxy.Namespace).Get(ctx, proxy.Name, metav1.GetOptions{})
+			if err != nil {
+				return
+			}
+			if len(proxySVC.Status.LoadBalancer.Ingress) == 1 {
+				// once an ingress IP is provisioned, we check to make sure we can get a 200 OK from the admin API as well
+				ip := proxySVC.Status.LoadBalancer.Ingress[0].IP
+				resp, err := http.Get(fmt.Sprintf("http://%s:8001/services", ip))
+				if err != nil {
+					continue
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode == http.StatusOK {
+					ready = true
+					break
+				}
+			}
+		}
+	}
+
+	// if the proxy deployment is ready, expose the new container ports via a LoadBalancer Service
+	if ready {
+		// configure the provided container ports for a LB service
+		svc = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: corev1.ServiceSpec{
+				Type:     corev1.ServiceTypeLoadBalancer,
+				Selector: proxy.Spec.Selector.MatchLabels,
+			},
+		}
+		for _, containerPort := range containerPorts {
+			servicePort := corev1.ServicePort{
+				Protocol:   containerPort.Protocol,
+				Port:       containerPort.ContainerPort,
+				TargetPort: intstr.FromInt(int(containerPort.ContainerPort)),
+			}
+			svc.Spec.Ports = append(svc.Spec.Ports, servicePort)
+		}
+		svc, err = cluster.Client().CoreV1().Services(ProxyDeploymentNamespace).Create(ctx, svc, metav1.CreateOptions{})
+		if err != nil {
+			return
+		}
+
+		// wait for the LB service to be provisioned
+		provisioned := false
+		timeout := time.Now().Add(serviceWait)
+		for timeout.After(time.Now()) {
+			svc, err = cluster.Client().CoreV1().Services(ProxyDeploymentNamespace).Get(ctx, name, metav1.GetOptions{})
+			if err != nil {
+				return
+			}
+			if len(svc.Status.LoadBalancer.Ingress) > 0 {
+				ing := svc.Status.LoadBalancer.Ingress[0]
+				if ip := ing.IP; ip != "" {
+					provisioned = true
+					break
+				}
+			}
+			time.Sleep(waitTick)
+		}
+		if !provisioned {
+			err = fmt.Errorf("load balancer service for deployment %s did not provision successfully within %s", name, serviceWait)
+		}
+	} else { // if the deployment didn't become ready within the timeout period, bail out there's nothing more we can do.
+		err = fmt.Errorf("deployment not ready after %s", timeout)
+	}
+
+	// FIXME: there appears to be a race condition that can occur in kong upstream if the Admin API is restarted and then updated
+	//        too quickly. Effectively the API will return 200 OK but it's not actually ready to accept updates. Given a few seconds
+	//        this will settle so as a workaround we're sleeping here, but we need to dig deeper into this problem to actually fix it.
```

Comment:
```suggestion
	//        this will settle so as a workaround we're sleeping here, but we need to dig deeper into this problem to actually fix it.
	//        See: https://github.com/Kong/kubernetes-ingress-controller/issues/941
```